### DISCUSSION TITLE
Scroll restoration?

### DISCUSSION THREAD

**Discussion by Darren120** [petyosi/react-virtuoso#1008] [Q&A]
How do I achieve scroll restoration? 
When I navigate away, then go back the scroll resets
is it just me or is it happening to anyone else?
i am using nextjs13

**Comment by pavel-karpovich** [comment]
https://virtuoso.dev/virtuoso-api/interfaces/VirtuosoProps/#restorestatefrom
You need to pass `ref` to Virtuoso component in order to get `VirtuosoHandle`. `VirtuosoHandle` exposes method `getState` which returns `StateSnapshot` which you will later provide to `restoreStateFrom` prop. And don't forget to update your `key` property to trigger remount if your component don't unmount by itself when you need to restore the state.
